metrics-operator/controllers/common/analysis/analysis_evaluator.go

```diff
@@ -0,0 +1,53 @@
+package analysis
+
+import (
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+)
+
+type AnalysisEvaluator struct {
+	ObjectiveEvaluator IObjectiveEvaluator
+}
+
+func NewAnalysisEvaluator(o IObjectiveEvaluator) AnalysisEvaluator {
+	return AnalysisEvaluator{
+		ObjectiveEvaluator: o,
+	}
+}
+
+func (ae *AnalysisEvaluator) Evaluate(values map[string]string, ad *v1alpha3.AnalysisDefinition) types.AnalysisResult {
+	result := types.AnalysisResult{
+		ObjectiveResults: make([]types.ObjectiveResult, 0, len(ad.Spec.Objectives)),
+	}
+
+	keyObjectiveFailed := false
+	for _, objective := range ad.Spec.Objectives {
+		// evaluate a single objective and store it's result
+		objectiveResult := ae.ObjectiveEvaluator.Evaluate(values, &objective)
+		result.ObjectiveResults = append(result.ObjectiveResults, objectiveResult)
```

Comment:
since we know the object count let's directly access the element via the index returned by the `range` operator and remove append to have a 5x perf improvement :D 

https://medium.com/@antoharyanto/serious-impact-slice-in-golang-for-speed-and-memory-usage-c0997f44eb76